client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -949,6 +951,12 @@ public final void fieldCapsAsync(FieldCapabilitiesRequest fieldCapabilitiesReque
             FieldCapabilitiesResponse::fromXContent, listener, emptySet());
     }
 
+    /**
+     * @deprecated If creating a new HLRC ReST API call, consider creating new actions instead of reusing server actions. The Validation
+     * layer has been added to the ReST client, and requests should extend
+     * {@link org.elasticsearch.protocol.xpack.common.ValidationException} instead of `ActionRequest`.
```

Comment:
the new recommended way requires using classes that sit under xpack packages?

Reply:
good catch, i should be moving these to a non xpack, common place. Ill fix that.

Reply:
@andrershov this point makes me wonder if we should do away with protocol altogether once we fix the existing x-pack classes, and just put all of these request/responses directly in the HLRC. Thoughts cc @javanna ?

Reply:
looking at this more, I think it probably makes sense to just move them now to HLRC so we can also put the new Validatable in HLRC too. this way its not *only* x-pack classes.

Reply:
I think that anything that is not under xpack packages is fine, I don't mind whether that's a new oss library or within the high-level client directly.

Reply:
For now Im going to move them into the same jar. If we find a need to put them into a new jar, we can always go that route. Id rather stick to simpler and then do it when we need it. Ive done this in d0ca91cd2871464cc160672a79f24b2006080d40